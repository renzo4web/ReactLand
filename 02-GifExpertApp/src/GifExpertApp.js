import { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <div className="font-display bg-gray-200 container-lg flex flex-col items-center	min-h-screen">
      <h1 className="text-center text-5xl font-bold my-4">Gift Experts</h1>
      <AddCategory setCategories={setCategories} />
      <ul className="w-full">
        {categories.map((cats) => (
          <GifGrid key={cats} category={cats} />
        ))}
      </ul>
    </div>
  );
};

export default GifExpertApp;
