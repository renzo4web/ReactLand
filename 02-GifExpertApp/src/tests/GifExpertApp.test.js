import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import GifExpertApp from "../GifExpertApp";

describe("Test main <GifExpertApp />", () => {
  test("should render the component ", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should  display a list of categories", () => {
    const categories = ["rick", "morty"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
