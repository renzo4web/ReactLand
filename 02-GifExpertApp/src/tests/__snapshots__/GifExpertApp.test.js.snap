// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test main <GifExpertApp /> should render the component  1`] = `
<div
  className="font-display bg-gray-200 container-lg flex flex-col items-center	min-h-screen"
>
  <h1
    className="text-center text-5xl font-bold my-4"
  >
    Gift Experts
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <ul
    className="w-full"
  />
</div>
`;
